### PR TITLE
Bob 69 select UI to select what will be built

### PR DESCRIPTION
**Pull request by FurkanUses**
It works

### CODE REVIEW

File: Assets/Scripts/SpawnTower.cs [hunk 1]
```diff
@@ -21,28 +21,42 @@ private void Start()
...
-                                    && SpawnObject.AllowedToSpawn(tower, _objSpawnPlace.position + _displacement)
-                                    && _myTime > _nextPlacement)
+        if (Input.GetKey(KeyCode.E))
         {
-            ResourceManager.UseResource(neededResources);
-            _nextPlacement = _myTime + placementRate;
-            SpawnObject.Spawn(tower, _objSpawnPlace.position + _displacement, _objSpawnPlace.rotation);
-            _nextPlacement -= _myTime;
-            _myTime = 0F;
+            Debug.Log("E");
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/SpawnTower.cs:36]: remove debug message

File: Assets/Scripts/SpawnTower.cs [hunk 2]
```diff
@@ -21,28 +21,42 @@ private void Start()
...
+            if (canPlaceTower())
+            {
+                Debug.Log("placeTower");
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/SpawnTower.cs:39]: remove debug message

File: Assets/Scripts/SpawnTower.cs [hunk 3]
```diff
@@ -21,28 +21,42 @@ private void Start()
...
+        return SpawnObject.AllowedToSpawn(_towerToSpawn, _objSpawnPlace.position + _displacement) && ResourceManager.HasResources(neededResources) && _myTime > _nextPlacement;
+    }
+
+    public void setTower(GameObject tower)
+    {
+        _towerToSpawn = tower;
+    }
+
     //TODO solve in NecessaryResources
     private Dictionary<string, int> createDic(NecessaryResources neededResources)
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/SpawnTower.cs:61]: unrelated to you  - but it seems easier to just refactor ResourceManger 
can you update the TODO?

File: Assets/Scripts/UI/HotbarController.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    private void setSlotSprites()
+    {
+        int i = 0;
+        foreach (Transform child in GameObject.Find("Hotbar").transform)
+        {
+            child.Find("ItemSprite").GetComponent<Image>().sprite = towers[i].transform.Find("Waffenlauf").GetComponent<SpriteRenderer>().sprite;
+            i++;
+        };
+    }
+}
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/UI/HotbarController.cs:43]: eof nl

**Review by SomethingWittyWithATechAspect — CHANGES_REQUESTED** [review]
Never ran code

Also Tests missing

File: Assets/Scripts/SpawnTower.cs (hunk 1 above)
Comment (FurkanUses) [reply Assets/Scripts/SpawnTower.cs:36]: done

File: Assets/Scripts/SpawnTower.cs (hunk 2 above)
Comment (FurkanUses) [reply Assets/Scripts/SpawnTower.cs:39]: done

File: Assets/Scripts/SpawnTower.cs (hunk 3 above)
Comment (FurkanUses) [reply Assets/Scripts/SpawnTower.cs:61]: no

File: Assets/Scripts/UI/HotbarController.cs (hunk 4 above)
Comment (FurkanUses) [reply Assets/Scripts/UI/HotbarController.cs:43]: done

**Comment by FurkanUses** [comment]
I know test are missing, I will add them later

File: Assets/Scripts/SpawnTower.cs (hunk 3 above)
Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/SpawnTower.cs:59]: Mean but fine

**Review by SomethingWittyWithATechAspect — APPROVED** [review]

**Comment by FurkanUses** [comment]
made new branch for merging

**Pull request closed without merging**